src/ch09-03-to-panic-or-not-to-panic.md

```diff
@@ -0,0 +1,487 @@
+        }
+
+        Guess {
+            value
+        }
+    }
+
+    pub fn value(&self) -> u32 {
+        self.value
+    }
+}
+```
+
+```rust
+pub struct Guess {
+    value: u32,
+}
+
+impl Guess {
+    pub fn new(value: u32) -> Guess {
+        if value < 1 || value > 100 {
+            panic!("Valor de chute deve ser entre 1 e 100, recebi {}.", value);
+        }
+
+        Guess {
+            value
+        }
+    }
+
+    pub fn value(&self) -> u32 {
+        self.value
+    }
+}
+```
+
+<span class="caption">Listing 9-9: A `Guess` type that will only continue with
+values between 1 and 100</span>
+
+<span class="caption">Lista 9-9: Um tipo `Guess` que somente funciona com valores
+entre 1 e 100</span>
+
+First, we define a struct named `Guess` that has a field named `value` that
+holds a `u32`. This is where the number will be stored.
+
+Primeiro, definimos uma struct chamada `Guess` que tem um campo chamado `value`
+que guarda um `u32`. Isso é onde o número vai ser guardado.
+
+Then we implement an associated function named `new` on `Guess` that creates
+instances of `Guess` values. The `new` function is defined to have one
+parameter named `value` of type `u32` and to return a `Guess`. The code in the
+body of the `new` function tests `value` to make sure it’s between 1 and 100.
+If `value` doesn’t pass this test, we make a `panic!` call, which will alert
+the programmer who is writing the calling code that they have a bug they need
+to fix, because creating a `Guess` with a `value` outside this range would
+violate the contract that `Guess::new` is relying on. The conditions in which
+`Guess::new` might panic should be discussed in its public-facing API
+documentation; we’ll cover documentation conventions indicating the possibility
+of a `panic!` in the API documentation that you create in Chapter 14. If
+`value` does pass the test, we create a new `Guess` with its `value` field set
+to the `value` parameter and return the `Guess`.
+
+Então nós implementamos uma função associada chamada `new` em `Guess` que cria
+instâncias de valores `Guess`. A função `new` é definida a ter um parâmetro
+chamado `value` de tipo `u32` e retornar um `Guess`. O código no corpo da função
+`new` testa para ter certeza que `value` está entre 1 e 100. Se `value` não passa
+nesse teste, fazemos uma chamada a `panic!`, que vai alertar ao programmador que
+está escrevendo o código chamando a função que ele têm um bug que precisa ser 
+corrigido, porque criar um `Guess` com um `value` fora desses limites violaria
+o contrato em que `Guess::new` se baseia. As condições em que `Guess::new` pode 
+entrar em pânico devem ser discutidas na sua documentação da API voltada ao público;
+nós cobriremos convencções de documentação indicando a possibilidade de um `panic!`
+na documentação de API que você cria no Capítulo 14. Se `value` de fato passa no
+teste, criamos um novo `Guess` com o campo `value` preenchido com o parâmetro
+`value` e retornamos o `Guess`.
```

Comment:
- ... que vai alertar ao ~programmador~ **programador**...
- ... chamando a função que ele ~têm~ **tem** um bug que precisa ser corrigido...
- ... nós cobriremos ~convencções~ **convenções** de documentação...
- Sugestão: no Capítulo 14, nós cobriremos convenções de documentação indicando a possibilidade de um `panic!` na documentação de API que você cria ~no Capítulo 14~. (Achei melhor colocar no início, porque achei estranho esse "that you create in Chapter 14").